File: cs50-x/week-2/ps2-readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int checkpunct(char c);
void printResult(int grade);

int main(void){
    int letters = 0, words = 0, sentences = 0;
    string text = get_string("Text: ");

    for(int i = 0, length = strlen(text); i < length; i++){
        if(isalpha(text[i])){
            letters++;
        }else if(isblank(text[i])){
            words++;
        }else if(checkpunct(text[i])){
            sentences++;
            if(i == length - 1){
                words++;
            }
        }
    }

    float index = 0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8;

    printResult((int)round(index));
}

int checkpunct(char c){
    switch(c){
        case '.':
            return 1;
        break;
        case '!':
            return 1;
        break;
        case '?':
            return 1;
        break;
        default:
            return 0;
        break;
    }
}

void printResult(int grade){
    if(grade < 1){
        printf("Before Grade 1\n");
    }else if(grade >= 16){
        printf("Grade 16+\n");
    }else{
        printf("Grade %i\n", grade);
    }
}

File: cs50-x/week-6/ps6-sentimental-readability/readability.py
text = input("Text: ")

letters = 0
words = 0
sentences = 0
for c in text:
    if c.isalpha():
        letters += 1
    elif c == '.' or c == '!' or c == '?':
        sentences += 1
    elif c == ' ':
        words += 1

words += 1

l = letters / words * 100
s = sentences / words * 100

index = 0.0588 * l - 0.296 * s - 15.8
index = round(index)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade", index)